torch/csrc/jit/serialization/export_bytecode.cpp

```diff
@@ -1,23 +1,322 @@
+  std::vector<Method> methods;
+  // Use DFS on IValue's to traverse dependencies of module._ivalue and
+  // add all setstate/getstates to initial stack.
+  std::vector<ObjectPtr> ivalue_stack;
+  ivalue_stack.emplace_back(obj);
+  while (!ivalue_stack.empty()) {
+    ObjectPtr cur = ivalue_stack.back();
+    ivalue_stack.pop_back();
+    auto type = cur->type();
+    Function* setstate = type->findMethod("__setstate__");
+    Function* getstate = type->findMethod("__getstate__");
+    if (getstate && setstate) {
+      if (setstate->isGraphFunction()) {
+        methods.emplace_back(cur, setstate);
+      }
+      if (getstate->isGraphFunction()) {
+        methods.emplace_back(cur, getstate);
+      }
+    } else {
+      for (size_t i = 0, n = type->numAttributes(); i < n; ++i) {
+        IValue field = cur->getSlot(i);
+        if (field.isObject()) {
+          ivalue_stack.emplace_back(field.toObject());
+        }
+      }
+    }
+  }
+  return methods;
 }
 
-void BytecodeExportSet::update(const c10::QualifiedName& qn, bool toplevel) {
-  items_.at(qn).toplevel = toplevel;
+// NOTE: order of functions returned will be:
+// 1. functions originated from the methods passed in will be first
+// 2. All the dependent functions will come afterwards.
+// This order is meaningful because currently mobile Module looks up
+// methods with linear search.
+std::vector<std::unique_ptr<GraphFunction>> inlineFunctions(
+    const std::vector<Method>& initial_methods,
+    bool incl_dependent_functions) {
+  std::set<std::pair<std::string, Function*>> visited;
+  std::deque<Method> stack;
+  std::copy(
+      initial_methods.begin(),
+      initial_methods.end(),
+      std::back_inserter(stack));
+  std::vector<std::unique_ptr<GraphFunction>> inlined_functions;
+  while (!stack.empty()) {
+    Method cur = stack.front();
+    stack.pop_front();
+    auto tup = std::make_pair(
+        cur.owner()._ivalue()->type()->name()->qualifiedName(),
+        &cur.function());
+    if (visited.find(tup) != visited.end()) {
+      continue;
+    }
+    visited.insert(tup);
+    const auto& f = toGraphFunction(cur.function());
+    auto graph = f.graph()->copyUnique();
+    Inline(*graph);
+    c10::QualifiedName qn(*cur.owner()._ivalue()->type()->name(), f.name());
+
+    if (incl_dependent_functions) {
+      std::unordered_set<c10::string_view> called_method_names;
+      auto nodes = findAllNodes(*graph, c10::prim::CallMethod, true);
+      for (Node* node : nodes) {
+        if (auto iface = node->input(0)->type()->castRaw<InterfaceType>()) {
+          const FunctionSchema* schema = iface->getMethod(node->s(attr::name));
+          called_method_names.insert(schema->name());
+        }
+      }
+
+      for (const auto& submodule : cur.owner().modules()) {
+        for (const auto& m : submodule.get_methods()) {
+          if (called_method_names.find(m.function().qualname().name()) !=
+              called_method_names.end()) {
```

Comment:
@qihqi Could we put this in a separate function? (like vector<> findInterfaceMethods()).

Reply:
done.